### PR TITLE
Wrap tests in pester blocks

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

Wrap tests in pester blocks to prepare for pesterv5

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: test/powershell/Language/Classes/scripting.Classes.NestedModules.tests.ps1 [hunk 1]
```diff
@@ -54,72 +53,74 @@ Describe 'NestedModules' -Tags "CI" {
             'class A { [string] foo() { return "A"} }',
             'class B { [string] foo() { return "B"} }'
         ) -Content 'class C { [string] foo() { return "C"} }'
+    }
+
+    AfterAll {
+        $env:PSModulePath = $originalPSModulePath
+        Get-Module @('ABC', 'NoRoot', 'WithRoot') | Remove-Module
```
Comment (iSazonov) [inline test/powershell/Language/Classes/scripting.Classes.NestedModules.tests.ps1:60]:
```suggestion
        Remove-Module 'ABC', 'NoRoot', 'WithRoot'
```

**Review by iSazonov — APPROVED** [review]
LGTM with one minor comment.

@xtqqczze Thanks! This helps us to move to Pester 5.0.

File: test/powershell/Language/Classes/scripting.Classes.NestedModules.tests.ps1 (hunk 1 above)
Comment (vexx32) [reply test/powershell/Language/Classes/scripting.Classes.NestedModules.tests.ps1:60]: That emits an error if the module isn't found, which doesn't happen for the former. Either we need `-ErrorAction SilentlyContinue` or we should keep the original version.

Comment (iSazonov) [reply test/powershell/Language/Classes/scripting.Classes.NestedModules.tests.ps1:60]: I agree to keep original.

**Comment by xtqqczze** [comment]
@iSazonov this was WIP, but I ran into some issues with multiple `Describe` blocks, so I will not add any more commits. This PR is ready to merge.

**Comment by iSazonov** [comment]
@xtqqczze If you are going to prepare all test to being Pester v5 compatible  it makes sense to open a meta issue where list all test files and their status.

**Comment by xtqqczze** [comment]
meta issue opened for wider pester v5 disussion #12816

**Pull request merged**